KarhooUISDKTests/TestCases/Screens/AddressScreen/KarhooAddressPresenterSpec.swift

```diff
@@ -206,18 +206,27 @@ class KarhooAddressPresenterSpec: XCTestCase {
      *  Then:   That text should be passed on to the search
      */
     func testSearch() {
+        testObject = KarhooAddressPresenter(
+            preferredLocation: mockLocation,
+            addressMode: mockAddressMode,
+            selectionCallback: { self.mockLocationInfoResult = $0 },
+            searchProvider: mockAddressSearchProvider,
+            userLocationProvider: mockUserLocationProvider,
+            addressService: mockAddressService,
+            analytics: mockAppAnalytics,
+            recentAddressProvider: mockRecentAddressProvider,
+            searchDelay: 0
+        )
+
         testObject.set(view: mockAddressView)
 
         mockUserLocationProvider.lastKnownLocation = TestUtil.getRandomLocation()
         let searchText = "Some text"
         testObject.search(text: searchText)
-        let expectation = XCTestExpectation()
 
-        DispatchQueue.global().asyncAfter(deadline: .now() + searchDelay*2) { [weak self] in
-            XCTAssertEqual(self?.mockAddressSearchProvider.searchString, searchText)
-            expectation.fulfill()
+        DispatchQueue.global(qos: .background).async {
+            XCTAssertEqual(self.mockAddressSearchProvider.searchString, searchText)
```

Comment:
The assert is throwing this error: `XCTAssertEqual failed: ("nil") is not equal to ("Optional("Some text")")`. The  `mockAddressSearchProvider.searchString` appears to be nil